scripts/check-license-compliance.js

```diff
@@ -0,0 +1,103 @@
+#! /usr/bin/env node
+'use strict';
+/* eslint no-console: [ "error", { allow: [ "info", "warn", "error", "table" ] } ] */
+const { init } = require('license-checker'),
+      process = require('process'),
+      util = require('util'),
+      initLicenseChecker = util.promisify(init),
+      packageDirectory = require('pkg-dir');
+
+const ALLOWED_LICENSES = [
+   'MIT',
+   'ISC',
+   'Apache-2.0',
+   '(MIT OR Apache-2.0)',
+   'Public Domain',
+   'CC-BY-3.0',
+];
+
+(async () => {
+   const projectDirectoryPath = await packageDirectory(),
+         projectConfigPath = `${projectDirectoryPath}/.license-checker.cjs`,
+         licenseRecord = {};
+
+   let allowList = [],
+       moduleInfoReport,
+       unsupportedLicenses;
+
+   console.info('Project package.json located at path:', projectDirectoryPath);
+
+   try {
+      // eslint-disable-next-line global-require
+      const config = require(projectConfigPath);
+
+      allowList = config.allowList;
+
+      console.info(`Using configured allow-list at path ${projectConfigPath}`);
+      console.info('Allowing additional licenses:', allowList.join(', '));
+   } catch(e) {
+      console.error(e);
+      console.info('A config for license checker was not provided or was not found.');
+   }
+
+   try {
+      moduleInfoReport = await initLicenseChecker({
+         start: process.cwd(),
+      });
+   } catch(e) {
+      console.error('Could not initialize license-checker', e);
+      process.exitCode = 1;
+      return;
+   }
+
+   Object
+      .keys(moduleInfoReport)
+      .forEach((moduleName) => {
+         const licenses = moduleInfoReport[moduleName].licenses;
+
+         if (!licenses) {
+            return;
+         }
+
+         // normalize licenses since license-checker returns either a string or array
+         // of the license information
+         if (Array.isArray(licenses)) {
+            moduleInfoReport[moduleName].licenses.forEach((license) => {
+               licenseRecord[moduleName] = {
+                  module: moduleName,
+                  license: license,
+                  ...moduleInfoReport[moduleName],
+               };
+            });
+
+            return;
+         }
+
+         licenseRecord[moduleName] = {
+            module: moduleName,
+            license: licenses,
+            ...moduleInfoReport[moduleName],
+         };
+      });
+
+   console.info(`This project has ${Object.values(licenseRecord).length} licenses.`);
+
+   unsupportedLicenses = Object.values(licenseRecord).filter((record) => {
+      return !ALLOWED_LICENSES.includes(record.license) && !allowList.includes(record.license);
```

Comment:
If a project supplies a list of permitted licenses, should we keep checking the default license list?

Reply:
I was imaging the licenses passed in would be _in addition_ to default set declared inside the tool.